pymc4/inference/sampling.py

```diff
@@ -166,12 +166,13 @@ def build_logp_and_deterministic_functions(
         )
     if state is not None and observed is not None:
```

Comment:
This one can also be removed since there is a check in evaluate_model but the exception there is not very intuitive for me.